example/offload-template/CMakeLists.txt

```
@@ -0,0 +1,53 @@
+#------------------------------- -*- cmake -*- -------------------------------#
+# Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+#-----------------------------------------------------------------------------#
+# Minimal Geant4 application with Celeritas offloading capabilities
+#-----------------------------------------------------------------------------#
+cmake_minimum_required(VERSION 3.10)
+cmake_policy(VERSION 3.12...3.30)
+project(offload-template)
+
+#-----------------------------------------------------------------------------#
+# Find packages
+find_package(Celeritas 0.5)
+find_package(Geant4 REQUIRED)
+
+if(NOT Celeritas_FOUND)
+  message(FATAL_ERROR
+  "No Celeritas installation found.\n"
+  "Export Celeritas_ROOT=/path/to/celeritas/install or use CMAKE_INSTALL_PREFIX")
+endif()
+
+#-----------------------------------------------------------------------------#
+# Set up sources, include directories, and libraries
+set(_sources
+  src/ActionInitialization.cc
+  src/Celeritas.cc
+  src/DetectorConstruction.cc
+  src/EventAction.cc
+  src/PrimaryGeneratorAction.cc
+  src/RunAction.cc
+  src/SensitiveDetector.cc
+)
+
+set(_includes
+  ${CMAKE_CURRENT_SOURCE_DIR}/src
+  ${Geant4_INCLUDE_DIR}
+  ${Celeritas_ROOT}/include
```

```suggestion
file(GLOB sources ${PROJECT_SOURCE_DIR}/src/*.cc)
file(GLOB headers ${PROJECT_SOURCE_DIR}/include/*.hh)
```
so that they do not need to add an additional list of codes in the cmake file.

---

No, globbing source is bad practice. The includes here are just directories, and in fact the `Geant4_INCLUDE_DIR` and `Celeritas_ROOT` lines should be removed since they're propagated by the CMake library targets.

---

Yes. I avoided globbing. Good catch on the include dirs, remnants from a poor copy-paste, since the `find_package` should already provide those.